src/Glpi/Inventory/Inventory.php

```diff
@@ -331,16 +332,38 @@ function ($property_name) {
 
             $this->data = $data;
 
+            //process itemtype definition from rules engine
+            $itemtype_string = $this->metadata['itemtype'] ?? 'Computer';
+            $main_itemtype = new Itemtype(
+                new CommonDBTM(),
```

Comment:
Would it be possible to adapt the code to pass a null value instead of a `CommonDBTM` instance? It seems weird to me to instanciate this class, as it is some kind of an abstract class.

Reply:
It's used at some places; but maybe can it be replaced, I'll check.

Reply:
Done